File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> largestValues(TreeNode* root) {
        vector<int> list;
        if(root==NULL) return list;
        queue<TreeNode*> queue;
        queue.push(root);
        queue.push(NULL);
        int max = INT_MIN;
        while(!queue.empty()){
            TreeNode* currNode = queue.front();
            queue.pop();
            if(currNode == NULL){
                list.push_back(max);
                max = INT_MIN;
                if(!queue.empty()){
                    queue.push(NULL);
                }
            }
            else{
                max =  std::max(currNode->val , max);
                if(currNode->left != NULL) queue.push(currNode->left);
                if(currNode->right != NULL) queue.push(currNode->right);
            }
        }
        return list;
    }
};

File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<int> LargestValues(TreeNode root) {
        List<int> list = new List<int>();
        if(root==null) return list;
        Queue<TreeNode> queue = new Queue<TreeNode>();
        queue.Enqueue(root);
        queue.Enqueue(null);
        int max = int.MinValue;
        while(queue.Count > 0){
            TreeNode currNode = queue.Dequeue();
            if(currNode == null){
                list.Add(max);
                max = int.MinValue;
                if(queue.Count > 0){
                    queue.Enqueue(null);
                }
            }
            else{
                max = Math.Max(currNode.val , max);
                if(currNode.left != null) queue.Enqueue(currNode.left);
                if(currNode.right != null) queue.Enqueue(currNode.right);
            }
        }
        return list;
    }
}